### PR TITLE
Add Animation Transition

### PR DESCRIPTION
**Pull request by sanjidbillah**
I have added animation transition. <code>ScaleTransition</code> as default.
If i need change anything let me know.
fix for #48

### CODE REVIEW

**Comment by minikin** [comment]
@sanjidbillah awesome, thank you very much for your contribution! 
I'll check it in detail asap.

**Comment by prateekmedia** [comment]
@sanjidbillah Nice Try, but it doesn't work, when setting it to fade transition I am still getting the same:

https://user-images.githubusercontent.com/41370460/153723498-25531d82-1e14-40f5-a52d-24566e37508b.mp4

**Comment by prateekmedia** [comment]
This is what I wanted and expected when I said fade transition:

https://user-images.githubusercontent.com/41370460/153723627-96e6860e-1195-4e5e-b49f-f8a2a5c80ecb.mp4

**Comment by prateekmedia** [comment]
@minikin My platform is linux and the code I used to achieve Fade transition of the popover widget is [here](https://github.com/gtk-flutter/popover)

Although Its not maintainable in the long run and also I have used a Hack.

To achieve it by setting initial scale to 1 so it doesnt scale and then using FadeTransition as the parent of the popover.

**Comment by prateekmedia** [comment]
@sanjidbillah Also I used below code:
```dart
showPopover(
      context: context,
      barrierColor: Colors.transparent,
      shadow: [
        BoxShadow(
          color: context.borderColor,
          blurRadius: 6,
        ),
      ],
      bodyBuilder: (_) => bodyHere,
      width: 200,
      height: null,
      backgroundColor: Theme.of(context).cardColor,
    )
```

**Comment by sanjidbillah** [comment]
@prateekmedia  I used same <code>FadeTransition</code> widget.
Did you check with my code? i already check it works.
If you try with my code then give me the code i will check it.

You gave me your code not my example code.

**Comment by prateekmedia** [comment]
@sanjidbillah You ARE having a fade transition but I was not talking about the backgrounds FadeTrandition but Fade transition of the popover.

See the first video you send me has Fade transition for the background but the popover widget which has all the contents still is popping out and using ScaleTransition.

**Comment by prateekmedia** [comment]
In short I wanted the popover to not scale but fade in when coming on screen.

**Comment by sanjidbillah** [comment]
@prateekmedia  got it. If @minikin allow me to change ScaleTransition I will change it. I am waiting for code review response.

**Comment by prateekmedia** [comment]
I think he will if you are talking about fixing #48, regarding your current code If you feel it's useful then you can rename it to something like,

```dart
backgroundTransition: PopoverBackgroundTransition.fadeTransition,
```

**Comment by sanjidbillah** [comment]
@prateekmedia Right, But now i am thinking i will keep scale only for ScaleTransition.

**Comment by prateekmedia** [comment]
> @prateekmedia For popover.

Ok, but you should remove `fix for #48` from description as this is not the fix for it. Thanks.

**Comment by sanjidbillah** [comment]
It will be fix soon

**Comment by sanjidbillah** [comment]
@prateekmedia Please check i have fixed all transition.

**Comment by prateekmedia** [comment]
PR looks great but I have an issue.
Issue:
The transition for background and popover doesn't need to be same like I
can want to have background transitioning with fade but the popover as
Scale.

Probable solution:
Add `backgroundTransition` property to popover and set its default value to
fade, you may ask why fade, its because the package uses FadeTransition for
background by default so this will be breaking change if its Scale for
background.

TL;DR
Separate transition for background as
```dart
backgroundTransition: PopoverTransition.fadeTransition,
```

**Comment by prateekmedia** [comment]
I am saying about keeping new changes too for popover but separate the
backgroundTransition and transition, this way the both transitions can be
different as well and also the background transition which was fade by
default can be there.

You can wait for @minikin, this was just my take on how it would be useful
for a larger group of users.

**Comment by minikin** [comment]
@sanjidbillah @prateekmedia What I'd like to propose is to think about how we can implement generic abstraction about transition animation. And do not limit our users to [PopoverTransition](https://github.com/minikin/popover/pull/50/files#diff-54cf94cd8f905ab99e4a4a269f2d523531a8eb7ec0f6527a2beae7b06f75d5c6R2) type. We can provide `ScaleTransition ` as the default option.

**Comment by prateekmedia** [comment]
@minikin Something like a transitionBuilder property

**Comment by sanjidbillah** [comment]
@minikin  @prateekmedia  Please check new PR. I have added popoverBuilder now user can use any animation also i set <code>ScaleTransition</code> as default.
I keep PopoverTransition Type user can disable Transition by  <code>PopoverTransition.none</code> by default it will use <code>ScaleTransition</code>.

**Comment by prateekmedia** [comment]
@minikin Any update on this?

**Comment by prateekmedia** [comment]
@minikin Any reason why this PR is not merged?

**Comment by prateekmedia** [comment]
@minikin Can you review the changes and comment

**Comment by sanjidbillah** [comment]
I tried but i found a similar open issue [here](https://github.com/flutter/flutter/issues/89905).
@minikin any suggestion?

**Comment by minikin** [comment]
@sanjidbillah We just need to update an example with a more applicable custom transition, and we'll be ready to merge.

**Comment by sanjidbillah** [comment]
@minikin I think I am not good at documentation but I will try -_-
